src/main/java/com/laserfiche/repository/api/clients/impl/ApiClient.java

```diff
@@ -88,10 +93,250 @@ protected String toJson(Object object) {
+        List<String> ResponseContentType = httpResponse
+                .getHeaders()
+                .get("Content-Type");
+        for (int i = 0; i < ResponseContentType.size(); i++) {
+            if (ResponseContentType
+                    .get(i)
+                    .contains("json")) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    protected static void createApiException(HttpResponse<Object> httpResponse, ProblemDetails problemDetails) {
+        int statusCode = httpResponse.getStatus();
+        Map<String, String> headersMap = getHeadersMap(httpResponse.getHeaders());
+        throw ApiException.create(statusCode, headersMap, problemDetails, null);
+    }
+
+    protected static String beforeSend(String url, Map<String, String> headerParametersWithStringTypeValue,
+            HttpRequestHandler httpRequestHandler) {
+        String requestUrl;
+        Request customRequest = new RequestImpl();
+        BeforeSendResult beforeSendResult = httpRequestHandler.beforeSend(customRequest);
+        String authorizationValue = customRequest
+                .headers()
+                .get("Authorization");
+        if (authorizationValue != null) {
+            headerParametersWithStringTypeValue.put("Authorization", authorizationValue);
+        }
+        if (url.startsWith("http")) {
+            requestUrl = url;
+        } else {
+            String apiBasedAddress = getRepositoryEndpoint(beforeSendResult.getRegionalDomain());
+            requestUrl = combineURLs(apiBasedAddress, url);
+        }
+        return requestUrl;
+    }
+
+    private static boolean shouldThrowException(String requestMethod, String queryStringFields,
+            Collection<?> queryStringFieldList, String url, ApiException err) {
+        if (requestMethod.equals(
+                "HEAD")) {
+            return true;
+        }
+        if (queryStringFields != null && queryStringFieldList != null) {
+            return true;
+        }
+        if (requestMethod.equals(
+                "PATCH")) {
+            return true;
+        }
+        if (requestMethod.equals("PUT") && url.contains("template")) {
+            return true;
+        }
+        if (url.contains(
+                "Task") && err
+                .getProblemDetails()
+                .getTitle()
+                .contains("Error")) {
+            return true;
+        }
+        if (err
+                .getProblemDetails()
+                .getTitle()
+                .equals("Authentication failed (reason: Invalid service principal key. (6-45)).")) {
+            return true;
+        }
+        if (err
+                .getProblemDetails()
+                .getTitle()
+                .equals("The user account name or password is incorrect. [9010]")) {
+            return true;
+        }
+        return false;
+    }
+
+    protected static <TResponse> TResponse sendRequestParseResponse(
```

Comment:
added a lambda parse function for every API client except for 2 additional edge cases (i.e. getDocumentContentType and getDynamicFieldValues)
For those 2 edge cases:
added a lambda function as a helper function in the `ApiClient` class for the `getDynamicFields` API
removed lambda function for getDocumentType since it was already included in the `sendResponse` function